pkg/apis/pipeline/v1/matrix_types.go

```diff
@@ -86,7 +104,22 @@ func (cs Combinations) fanOut(param Param) Combinations {
 	return cs.distribute(param)
 }
 
-// distribute generates a new combination of Parameters by adding a new Parameter to an existing list of Combinations.
+// fanOutExplictCombinations handles the use case when there are only Matrix Include Parameters
+// and no Matrix Parameters to generate explicit combinations
+func (m *Matrix) fanOutExplictCombinations() Combinations {
+	var combinations Combinations
+	for i := 0; i < len(m.Include); i++ {
```

Comment:
for _, i := range(m.Include) {
		newCombination := make(Combination)
		for _, param := range i.Params {
			newCombination[param.Name] = param.Value.StringVal
		}
```suggestion
	for i := 0; i < len(m.Include); i++ {
```